source/extensions/load_balancing_policies/maglev/config.cc

```diff
@@ -15,24 +15,26 @@ Factory::create(OptRef<const Upstream::LoadBalancerConfig> lb_config,
                 const Upstream::PrioritySet& priority_set, Runtime::Loader& runtime,
                 Random::RandomGenerator& random, TimeSource&) {
 
-  const auto* typed_lb_config =
-      dynamic_cast<const Upstream::LoadBalancerConfigWrapper*>(lb_config.ptr());
-  auto typed_proto_config = typed_lb_config == nullptr
-                                ? OptRef<const MaglevLbProto>{}
-                                : typed_lb_config->typedProtoConfig<MaglevLbProto>();
+  const auto* typed_lb_config = dynamic_cast<const Upstream::TypedMaglevLbConfig*>(lb_config.ptr());
+
+  const auto* legacy_lb_config =
+      dynamic_cast<const Upstream::LegacyTypedMaglevLbConfig*>(lb_config.ptr());
 
   // Assume legacy config.
-  if (!typed_proto_config.has_value()) {
```

Comment:
You miss an assertion  ```ASSERT(typed_lb_confg != nullptr || legacy_lb_config != nullptr,
         "Invalid load balancing policy configuration for least request load balancer");``` which added to other extensions

Reply:
Although it's weird, but maglev and ring hash needn't that. 🤣 Because some historical reasons, these factories (maglev and ring hash) are also be used in legacy code path and `nullptr` will be used as `lb_config` argument.

We can only add that back after six month, when the legacy code path is removed completely.